### ISSUE TITLE
Broaden scope of full text search

### ISSUE THREAD

**Issue by jonrkarr** [KarrLab/datanator_rest_api#68]
- KEGG ortholog group ids
   - E.g., searching for `K11984` should return a Gene hit to the K11984 ortholog group
- EC numbers
   - E.g., searching for `1.1.1.1` should return Reactions hits with EC number 1.1.1.1

In my limited testing, searching by the following ids works: To confirm, are these fields included in the full text search?
- ChEBI ids
- KEGG compound ids
- UniProt ids

**Comment by lzy7071** [comment]
To include fields in full text search, one only needs to add field names in the `fields` parameter because the fields are already indexed in Elasticsearch.
I'll add them in the frontend code.

**Comment by jonrkarr** [comment]
Good to know. I didn't realize this was configurable from the frontend side. Thanks for taking care of this.

**Comment by lzy7071** [comment]
> Good to know. I didn't realize this was configurable from the frontend side. Thanks for taking care of this.

No problem. One does need to get a sample document to know what each field is actually named in the database. 
I changed field names to reflect the meta document schema for each category in https://github.com/KarrLab/datanator_frontend/commit/df69934bad89037ad673c7b2ce8430f6716d71ab. Reaction meta document will be changed soon because we are adding in BRENDA information.

**Comment by jonrkarr** [comment]
I added `ec-code` to the list of reaction fields to search over.

@lzy7071 How can I view sample documents to see the available fields?

**Comment by lzy7071** [comment]
> I added `ec-code` to the list of reaction fields to search over.
> 
> @lzy7071 How can I view sample documents to see the available fields?

One can use [/ftx/text_search/num_of_index/?query_message={}&index={}](http://api.datanator.info/ftx/text_search/num_of_index/?query_message=glucose&index=sabio_reaction_entries&from_=0&size=10&fields=product_names&fields=substrate_names&fields=enzyme_names
) to get see the available fields in documents in each `index`

**Comment by jonrkarr** [comment]
Example metabolite and reaction documents:
- **Metabolite:** http://api.datanator.info/ftx/text_search/num_of_index/?query_message=glucose&index=metabolites_meta&from_=0&size=10&fields=name
- **Reaction:** http://api.datanator.info/ftx/text_search/num_of_index/?query_message=1.1.1.1&index=sabio_reaction_entries&from_=0&size=10&fields=ec-code

@lzy7071 How can I view a sample gene document?

**Comment by lzy7071** [comment]
> Example metabolite and reaction documents:
> 
> * **Metabolite:** http://api.datanator.info/ftx/text_search/num_of_index/?query_message=glucose&index=metabolites_meta&from_=0&size=10&fields=name
> * **Reaction:** http://api.datanator.info/ftx/text_search/num_of_index/?query_message=1.1.1.1&index=sabio_reaction_entries&from_=0&size=10&fields=ec-code
> 
> @lzy7071 How can I view a sample gene document?

Our gene document is based on uniprot collection. However, the frontend requires a group-by-ko transformation. If one just wants to view the raw documents:

- Genes: [http://api.datanator.info/ftx/text_search/num_of_index/?query_message=dehydrogenase&index=protein&from_=0&size=10&fields=protein_name&fields=gene_name
](http://api.datanator.info/ftx/text_search/num_of_index/?query_message=dehydrogenase&index=protein&from_=0&size=10&fields=protein_name&fields=gene_name
)

**Comment by jonrkarr** [comment]
I cleaned up the fields used for full-text search. I think Yin Hoon's feedback is now addressed.

I removed these fields from the reaction search because the reaction documents don't appear to use these fields. If this is incorrect, let me know. I realize the reaction fields will need to be updated as you change the underlying organization.
- `enzyme_name`
- `enzymes`
- `enzymes.subunit.canonical_sequence`
- `gene_name`
- `ko_name`
- `name`
- `protein_name`
- `synonyms`
- `species`

**Comment by lzy7071** [comment]
> I cleaned up the fields used for full-text search. I think Yin Hoon's feedback is now addressed.
> 
> I removed these fields from the reaction search because the reaction documents don't appear to use these fields. If this is incorrect, let me know. I realize the reaction fields will need to be updated as you change the underlying organization.
> 
> * `enzyme_name`
> * `enzymes`
> * `enzymes.subunit.canonical_sequence`
> * `gene_name`
> * `ko_name`
> * `name`
> * `protein_name`
> * `synonyms`
> * `species`

I presume `enzyme_names` is still included for reaction document search? Other wise that sounds good to me. Thank you.
The default field names were only there as a suggestion of sorts. My intention was for the user to get some sample documents and decide which fields to include.

**Comment by jonrkarr** [comment]
Yes, I corrected `enzyme_name` --> `enzyme_names` (added missing `s`)
